lib/runtime/procs/fanin.js

```diff
@@ -130,11 +131,22 @@ class Input {
 
         if (timelessInput) { return timelessInput; }
 
+        function hasInterval(a) {
+            return a.mark && a.interval;
+        }
```

Comment:
The name of the function is misleading — it checks more than it says. Either rename to something like `isMarkWithInterval` or just inline it.